### DISCUSSION TITLE
set_my_commands usage example?

### DISCUSSION THREAD

**Discussion by Melistraza** [python-telegram-bot/python-telegram-bot#3800] [Q&A] [answered]
Is there any explanation how to define command from code source?

With [telegram.Bot.set_my_commands](https://docs.python-telegram-bot.org/en/stable/telegram.bot.html#telegram.Bot.set_my_commands)

Let's imagine we have simple bot
```python
from telegram.ext import Application
from telegram.ext import CommandHandler

def run_polling():
    application = Application.builder().token(TELEGRAM_TOKEN).build()
    application.add_handler(CommandHandler('start', start))
    application.run_polling()

if __name__ == "__main__":
    run_polling()
``` 

I have tried a lot of different approaches and tried to get `Bot` instance from `application` variable.
But it's doesn't work properly.

**Comment by clot27** [comment]
you are looking for [telegram.ext.Application.post_init](https://docs.python-telegram-bot.org/telegram.ext.application.html#telegram.ext.Application.post_init)

**Reply by Melistraza** [reply]
There no clear, good example, how to register command…

**Reply by harshil21** [reply, marked as answer]
You'll find that exact code snippet in https://docs.python-telegram-bot.org/en/stable/telegram.ext.applicationbuilder.html#telegram.ext.ApplicationBuilder.post_init
